fundingmanager.go

```diff
@@ -754,3 +796,37 @@ func (f *fundingManager) handleInitFundingMsg(msg *initFundingMsg) {
 	)
 	msg.peer.queueMsg(fundingReq, nil)
 }
+
+
+// processErrorGeneric sends a message to the fundingManager allowing it
+// to process the occurred generic error.
+func (f *fundingManager) processErrorGeneric(err *lnwire.ErrorGeneric,
+	peer *peer) {
+	f.fundingMsgs <- &fundingErrorMsg{err, peer}
+}
+
+
+// handleErrorGenericMsg process the error which was received from remote peer,
+// depends on the type of error we should do different clean up steps and
+// inform user about it.
+func (f *fundingManager) handleErrorGenericMsg(fmsg *fundingErrorMsg) {
+	if fmsg.err.ErrorID == lnwire.ErrorMaxPendingChannels {
+		peerID := fmsg.peer.id
+		chanID := fmsg.err.PendingChannelID
+
+		f.resMtx.RLock()
+		resCtx := f.activeReservations[peerID][chanID]
+		f.resMtx.RUnlock()
+
+		if err := resCtx.reservation.Cancel(); err != nil {
```

Comment:
In addition to canceling the reservation, the `resCtx` should also be removed from the `activeReservations` map.